tools/generator/Java.Interop.Tools.Generator.Importers/XmlApiImporter.cs

```diff
@@ -110,9 +110,14 @@ public static Field CreateField (XElement elem)
 
 			if (elem.Attribute ("managedName") != null)
 				field.Name = elem.XGetAttribute ("managedName");
-			else
+			else {
 				field.Name = TypeNameUtilities.StudlyCase (char.IsLower (field.JavaName [0]) || field.JavaName.ToLower ().ToUpper () != field.JavaName ? field.JavaName : field.JavaName.ToLower ());
 
+				// Preserve starting underscore if 2nd char is a number (removing it is an invalid C# name)
+				if (field.Name.Length > 0 && char.IsNumber (field.Name [0]) && field.JavaName.StartsWith ("_"))
```

Comment:
Why care about the *Java* side in this case?  `$` is also a valid Java identifier character, but not in C#, so what if we hit:

```java
class Example {
    public int $x;
}
```

(Yes, this compiles.  I have no idea what will happen with *binding*.  Would be interesting to find out.)

If `field.Name[0]` is not a valid "start identifier" character -- or `method[0]`, or `property[0]`, or *anything`[0]`* `[0]`! -- then we need to "escape"  it *somehow*, *probably* by prefixing with `_`.  And if `field.JavaName` is `$x`, then we *still* won't properly escape...which would be a problem.